src/app/Fake.BuildServer.TeamCity/TeamCity.fs

```diff
@@ -394,10 +403,15 @@ module TeamCity =
                     match description with
                     | Some d -> TeamCityWriter.sendOpenBlock tag.Name (sprintf "%s: %s" tag.Type d)
                     | _ -> TeamCityWriter.sendOpenBlock tag.Name tag.Type
+                | TraceData.CloseTag (tag, _, status) when status = TagStatus.Failed ->
```

Comment:
`| TraceData.CloseTag (tag, _, TagStatus.Failed)`?

Reply:
Wait...that's a think...

Reply:
Called "pattern matching" ;)